tests/model_selection/test_cross_validate.py

```diff
@@ -268,3 +273,128 @@ def test_happy_path_with_features(self, prefer_warm_inference_over_cold: bool) -
         }
 
         assert actual == expected
+
+    @pytest.mark.parametrize(
+        "ref_models,validate_ref_models,expected_metrics",
+        (
+            (
+                ["popular"],
+                False,
+                [
+                    {"model": "random", "i_split": 0, "precision@2": 0.5, "recall@1": 0.0, "intersection_popular": 0.5},
+                    {
+                        "model": "random",
+                        "i_split": 1,
+                        "precision@2": 0.375,
+                        "recall@1": 0.5,
+                        "intersection_popular": 0.75,
+                    },
+                ],
+            ),
+            (
+                ["popular"],
+                True,
+                [
+                    {"model": "popular", "i_split": 0, "precision@2": 0.5, "recall@1": 0.5},
+                    {"model": "random", "i_split": 0, "precision@2": 0.5, "recall@1": 0.0, "intersection_popular": 0.5},
+                    {"model": "popular", "i_split": 1, "precision@2": 0.375, "recall@1": 0.25},
+                    {
+                        "model": "random",
+                        "i_split": 1,
+                        "precision@2": 0.375,
+                        "recall@1": 0.5,
+                        "intersection_popular": 0.75,
+                    },
+                ],
+            ),
+            (
+                ["random"],
+                False,
+                [
+                    {"model": "popular", "i_split": 0, "precision@2": 0.5, "recall@1": 0.5, "intersection_random": 0.5},
+                    {
+                        "model": "popular",
+                        "i_split": 1,
+                        "precision@2": 0.375,
+                        "recall@1": 0.25,
+                        "intersection_random": 0.75,
+                    },
+                ],
+            ),
+            (
+                ["random"],
+                True,
+                [
+                    {"model": "popular", "i_split": 0, "precision@2": 0.5, "recall@1": 0.5, "intersection_random": 0.5},
+                    {"model": "random", "i_split": 0, "precision@2": 0.5, "recall@1": 0.5},
+                    {
+                        "model": "popular",
+                        "i_split": 1,
+                        "precision@2": 0.375,
+                        "recall@1": 0.25,
+                        "intersection_random": 0.75,
+                    },
+                    {"model": "random", "i_split": 1, "precision@2": 0.375, "recall@1": 0.25},
+                ],
+            ),
+            (
+                ["random", "popular"],
+                False,
+                [],
+            ),
+            (
+                ["random", "popular"],
```

Comment:
Everything is right except for this case (and the problem is not in the test itself but in the logic in `cross_validate`).
When we have all models passed as ref_models and we have `validate_ref_models` set to True, we need to have intersections calculated for all models on all others. Instead of this here we get all intersections dropped for all models.

Nice test cases selection btw, very useful for this case.